src/queue.js

```diff
@@ -81,6 +83,34 @@ export default class Queue extends EventEmitter {
       return Bluebird.resolve()
     }
 
+    const receiveMessages = (messagesResult = []) => {
+      const maxNumberOfMessages = self.options.concurrency > MAX_SQS_CONCURRENCY
+        ? MAX_SQS_CONCURRENCY
+        : self.options.concurrency
+
+      return Bluebird.resolve(self.options.sqs.receiveMessage({
+          QueueUrl: self.options.endpoint,
+          MaxNumberOfMessages: maxNumberOfMessages,
+          MessageAttributeNames: options.messageAttributeNames || ['All'],
+          AttributeNames: options.attributeNames || ['All'],
+        }).promise())
+        .get('Messages')
+        .then(coerce)
+        .then((messages) => {
+          const isMaxNumberOfMessages = messages.length >= MAX_SQS_CONCURRENCY
```

Comment:
Não sei se interpretei corretamente, mas acredito que só queremos executar a `receiveMessages` novamente caso o `messagesResult.length` seja maior que o `MAX_SQS_CONCURRENCY`, se for igual já podemos retornar pois já consultamos o concurrency desejado.

Minha sugestão seria.

```suggestion
          const isGreaterThenNumberOfMessages = messages.length > MAX_SQS_CONCURRENCY
```

Reply:
Na verdade essa condiçao só vai servir pra quando retornar 10 mensagens, pq o `messages` é o que é recebido do SQS.

@lucianopf vou pegar seu PR e acertar esses detalhes